lms/djangoapps/certificates/signals.py

```diff
@@ -118,8 +128,13 @@ def _listen_for_id_verification_status_changed(sender, user, **kwargs):  # pylin
     expected_verification_status = IDVerificationService.user_status(user)
     expected_verification_status = expected_verification_status['status']
     for enrollment in user_enrollments:
-        if grade_factory.read(user=user, course=enrollment.course_overview).passed:
-            if fire_ungenerated_certificate_task(user, enrollment.course_id, expected_verification_status):
+        if is_using_certificate_allowlist_and_is_on_allowlist(user, enrollment.course_id):
+            log.info(f'{enrollment.course_id} is using allowlist certificates, and the user {user.id} is on its '
+                     f'allowlist. Attempt will be made to generate an allowlist certificate. Id verification status '
+                     f'is {expected_verification_status}')
+            generate_allowlist_certificate_task(user, enrollment.course_id)
+        elif grade_factory.read(user=user, course=enrollment.course_overview).passed:
```

Comment:
Do you mind confirming my understanding here? `_listen_for_id_verification_status_changed` is called then a user verifies using the `LEARNER_NOW_VERIFIED` signal. Before this change, WL learners would have potentially not received their certs due to there being no grade.

This change separates the generic certificate path and checks from the Allowlist checks.

Are we slowly moving to a place where we can centralize what path is taken in a single method? It seems like the check we are doing here should be made on almost every evaluation of a learner's certificate needs.

Please reach out if my questions here are confusing. This comment does not block the PR.

Reply:
Thank you for your clarification.